### PR TITLE
SecurityContext : Avoid using deprecated APIs in Symfony 2.6+

### PR DESCRIPTION
**Pull request by notFloran**

### CODE REVIEW

**Comment by notFloran** [comment]
ping @nclsHart @martin-damien  @alioguz

**Comment by ghost** [comment]
:+1:

File: src/DependencyInjection/Compiler/SecurityContextPass.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        } elseif ($container->has('security.context')) {
+            $argument = new Reference('service_container');
+        } else {
+            return; // SecurityBundle is not configured
+        }
+
+        $definition = $container->findDefinition('hexanet_monolog_extra.logger.provider.user.symfony');
+        $definition->addArgument($argument);
+    }
+}
```
Comment (nclsHart) [inline src/DependencyInjection/Compiler/SecurityContextPass.php]: psr

Comment (notFloran) [inline src/DependencyInjection/Compiler/SecurityContextPass.php]: ack

File: src/Provider/User/SymfonyUserProvider.php [hunk 2]
```diff
@@ -17,17 +19,25 @@ class SymfonyUserProvider implements UserProviderInterface
...
-        $this->container = $container;
+        // BC layer for Symfony 2.5 and older
+        if ($tokenStorage instanceof ContainerInterface) {
+            $this->tokenStorage = $tokenStorage;
+
+            return;
+        }
+
+        if (null !== $tokenStorage && !$tokenStorage instanceof TokenStorageInterface) {
+            throw new \InvalidArgumentException(sprintf('The first argument of the %s constructor should be a Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface or a Symfony\Component\DependencyInjection\ContainerInterface or null.', __CLASS__));
```
Comment (nclsHart) [inline src/Provider/User/SymfonyUserProvider.php:37]: Maybe you could use `::class` instead of the FQCN.

**Comment by nclsHart** [comment]
:+1:

**Pull request merged**